annotate.py

```diff
@@ -85,34 +87,70 @@ def annotate_img(img_path, initials, size1=1, size2=1) :
     rdrawing=False
 
     print("\n Click on image to grab pixel coordinates.\n Press s to close image and exit program.\n Press r to write coordinates to file. \n    (you will be prompted for the PMT feature ID)")
-    
+
+
+
     while(1):
         cv2.imshow(filename,img)     ##keep displaying the image even if the user exits.
         k = cv2.waitKey(20) & 0xFF
         if(k == ord('s')):          ##if the s key is pressed, exit the loop.
-            #cv2.destroyWindow(filename)
+            print("Saving to: ",filename+".txt\n")
+            for i in range(len(coords[0])):
+                if(coords[0][i]!="N"):
+                    writestartingVal = f'{i:02}'
+
+                    print(filename+"\t"+writepmtID+"-"+writestartingVal+"\t"+str(coords[0][i])+"\t"+str(coords[1][i])+"\t"+initials+"\n")
+                    file.write("%s\t%s-%s\t%s\t%s\t%s\n" %(filename,writepmtID,writestartingVal, str(coords[0][i]), str(coords[1][i]),initials))
+                    coords[0][i]="N"
+                    coords[1][i]="N"  
             break
 
         
 ###############################Add line to text output###############################
         if(k==ord('f')):
            nextPMT = True
            pmtSelected = False
+           print("Saving to: ",filename+".txt\n")
+           
+           
+           for i in range(len(coords[0])):
+                if(coords[0][i]!="N"):
+                    writestartingVal = f'{i:02}' 
+                    print(filename+"\t"+writepmtID+"-"+writestartingVal+"\t"+str(coords[0][i])+"\t"+str(coords[1][i])+"\t"+initials+"\n")
+                    file.write("%s\t%s-%s\t%s\t%s\t%s\n" %(filename,writepmtID,writestartingVal, str(coords[0][i]), str(coords[1][i]),initials))
+                    coords[0][i]="N"
+                    coords[1][i]="N"  
            print("Ended recording for PMT ",str(pmtID)+". Record first feature for another by selecting a point and pressing r.")
-
-        elif(k==ord('x')):
-           print("Skipping feature #", startingVal)
+        elif(k==ord('n')):
            startingVal = startingVal+1
+           print("Ready to record", startingVal)
+
+        elif(k==ord('b')):
+            if(startingVal==0):
+                print("Error, can't decrement feature ID below 0.")
+            else:
+                startingVal = startingVal-1
+                print("Ready to record",startingVal)
 
         if(k==ord('r')):
             if(pmtSelected == True):
                 
-                writepmtID = pmtID.zfill(5)
-                writestartingVal = f'{startingVal:02}'   
+                #startingVal = startingVal+1
+                writestartingVal = f'{startingVal:02}'
+                writepmtID = pmtID.zfill(5)   
                 print("Recording ", writepmtID+"-"+writestartingVal, ix, "x", iy,"y\n")
-   
-                file.write("%s\t%s-%s\t%d\t%d\t%s\n" %(filename,writepmtID,writestartingVal, ix, iy,initials))
-                startingVal = startingVal+1   
```

Comment:
I would keep the automatic incrementing after recording, otherwise need an extra keystroke for each feature (`r` + `n`).

Reply:
readded automatic incrementing 👍